### DISCUSSION TITLE
Route isn't the same on refresh

### DISCUSSION THREAD

**Discussion by unpseudocomplique** [vuejs/router#2383] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/vue3-vite-router-starter-dnet26?file=src%2FApp.vue,src%2Frouter.js

### Steps to reproduce the bug

When app is running no issue when navigating between tabs.
But if you go on about page and hit refresh the page is first initialisated on / and then go to /about

<img width="494" alt="image" src="https://github.com/user-attachments/assets/26af662e-55e6-4d25-8ee5-26326222e58a">

### Expected behavior

Route be consistant on first load

### Actual behavior

Route changed after a few ms

### Additional information

I tryed to reproduce as little as possibile.
In the actual app drawer is always open when a user is connected but in a case where the user is a candidate that don't have a real access he should not be able to have the drawer. So on the route of his onboarding i have the meta hideDrawer but as you can see the route picked is not the right one at first

**Comment by posva** [comment, marked as answer]
This is because thi initial navigation is async: https://router.vuejs.org/guide/migration/#All-navigations-are-now-always-asynchronous
What you see is the `START_LOCATION` variable. You can change this behavior by doing `await router.isReady()` before `app.mount()`
